geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java

```diff
@@ -375,7 +379,8 @@ public void begin() {
    */
   public TXStateProxy beginJTA() {
     checkClosed();
-    TXId id = new TXId(this.distributionMgrId, this.uniqId.incrementAndGet());
+    TXId id = new TXId(this.distributionMgrId,
+        this.uniqId.updateAndGet(i -> i == Integer.MAX_VALUE ? 0 : i + 1));
```

Comment:
It might be better to have a method getNewTXUniqueId() to generate the next TXId?

Reply:
I have created new method called getNewTXId(), since it returns TXId object. Hope this is fine? I can change it to getNewTXUniqueID() if needed.